apps/server/src/modules/tool/service/context-external-tool.service.spec.ts

```diff
@@ -77,4 +80,44 @@ describe('ContextExternalToolService', () => {
 			});
 		});
 	});
+
+	describe('getContextExternalToolById is called', () => {
+		describe('when contextExternalToolId is given', () => {
+			const setup = () => {
+				const contextExternalTool: ContextExternalToolDO = contextExternalToolDOFactory.build();
+
+				return {
+					contextExternalToolId: contextExternalTool.id as string,
+				};
+			};
+
+			it('should call getContextExternalToolById()', async () => {
```

Comment:
If left like so the description should be "should call repo" or something like that. But it would be better to test for the return value here. As a rule of thumb: You only need to test for calls directly if they have side effects. Here, if repo.findById is not called, the return value changes. So it is enough to test for the return value.

This is also expressed by the point "stick to blackbox testing: think about the unit from the outside, not its inner workings." in our docs: https://hpi-schul-cloud.github.io/schulcloud-server/additional-documentation/nestjs-application/testing.html